### PR TITLE
adding none check for empty dataframes in system analysis

### LINKED ISSUE #18: Add None Check for Empty DataFrames in System Analysis
- Generated data frames sometimes result in None values while performing System Analysis. (E.g. If no Diacritic Error Found, then the generated data frame will throw an error when described.)

### PR DESCRIPTION
**Pull request by Muktan**
- [x] Tested code

Performed Error handling for System analysis.

### CODE REVIEW

**Comment by Muktan** [comment]
**Test case:** for input file having no error in hypothesis, w.r.t actual. 

Following is the Test CSV.
```
Actual_temp_model,Hypothesis_Greedy_temp_model
કેનેડાના પાંચ પ્રાંતે નિસર્ગોપચારકોને સારવાર કરવાની મંજૂરી આપી છે બ્રિટિશ કોલંબિયા મેનિટોબા નોવા સ્કોટિયા ઓન્ટારિયો અને સાસ્કાત્શેવન,કેનેડાના પાંચ પ્રાંતે નિસર્ગોપચારકોને સારવાર કરવાની મંજૂરી આપી છે બ્રિટિશ કોલંબિયા મેનિટોબા નોવા સ્કોટિયા ઓન્ટારિયો અને સાસ્કાત્શેવન
```

Following is the test result.
Screenshot from 2021-04-28 00-48-19

**Review by deepang17 — APPROVED** [review]
Seems Good

File: System Analysis/system_analysis.py [hunk 1]
```diff
@@ -13,15 +13,45 @@ def word_level(model_name, type):
...
+        # Hypothesis sentences for a particular decoding technique.
+        pred = list(Sentdata["Hypothesis_" + type + model_name].values)
+    except:
+        print(
+            "CSV does not have Hypothesis_"
+            + type
+            + model_name
+            + " attribute. please provide correct type and/or model_name"
+        )
+        print("skipping analysis...")
```
Comment (01-vyom) [inline System Analysis/system_analysis.py:53]: Skipping Analysis...

File: System Analysis/system_analysis.py [hunk 2]
```diff
@@ -13,15 +13,45 @@ def word_level(model_name, type):
...
-    # Hypothesis sentences for a particular decoding technique.
-    pred = list(Sentdata["Hypothesis_" + type + model_name].values)
+    try:
+        # Ground Truth sentences.
+        ground = list(Sentdata["Actual_" + model_name].values)
+    except:
+        print(
+            "CSV does not have Actual_"
+            + model_name
+            + " attribute. please provide correct model_name"
```
Comment (01-vyom) [inline System Analysis/system_analysis.py:38]: Please provide the correct model_name or check the name format of the .csv column

File: System Analysis/system_analysis.py [hunk 3]
```diff
@@ -13,15 +13,45 @@ def word_level(model_name, type):
...
+
+    try:
+        # Hypothesis sentences for a particular decoding technique.
+        pred = list(Sentdata["Hypothesis_" + type + model_name].values)
+    except:
+        print(
+            "CSV does not have Hypothesis_"
+            + type
+            + model_name
+            + " attribute. please provide correct type and/or model_name"
```
Comment (01-vyom) [inline System Analysis/system_analysis.py:51]: attribute`s`. Please provide correct type and/or model_name and check the name format of the .csv column

File: System Analysis/system_analysis.py [hunk 4]
```diff
@@ -862,41 +959,58 @@ def character_level(model_name, type):
...
+            + str(cons_df.head())
+            + "\n"
+            + str(cons_df.describe())
+            + "\n"
+        )
+        # print(sent)
+        analysis.append(sent)
+    except:
+        print(
+            "No sufficient data to perform some analysis. Skipping current, moving to next analysis."
```
Comment (01-vyom) [inline System Analysis/system_analysis.py:975]: Due to insufficient data, skipping some analysis and moving to the next analysis.

File: System Analysis/system_analysis.py [hunk 5]
```diff
@@ -13,15 +13,45 @@ def word_level(model_name, type):
...
 
     # CSV containing Hypothesis and ground truth sentences.
-    Sentdata = pd.read_csv("Eval/" + model_name + "_ALL_DECODING.csv")
+    try:
+        Sentdata = pd.read_csv("Eval/" + model_name + "_ALL_DECODING.csv")
+    except:
+        print(
+            "Unable to read "
+            + model_name
+            + "_ALL_DECODING.csv Check if it is available in Eval folder"
```
Comment (01-vyom) [inline System Analysis/system_analysis.py:22]: `c`heck

File: System Analysis/system_analysis.py (hunk 2 above)
Comment (Muktan) [reply System Analysis/system_analysis.py:38]: Replacing it with "CSV does not have a column with column name Actual_" + model_name + " which is required."

**Review by 01-vyom — APPROVED** [review]

**Pull request merged**